client_test.go

```diff
@@ -134,7 +134,6 @@ func TestMain(m *testing.M) {
 		}
 		if err := os.RemoveAll(defaultRoot); err != nil {
 			fmt.Fprintln(os.Stderr, "failed to remove test root dir", err)
-			os.Exit(1)
```

Comment:
This changes the behavior of testmain.

Reply:
We had some discussion about this previously in this PR. I suggested that there shouldn't be any reason to exit(1) here. Am I mistaken? What's the reason for an `Exit(1)` if the remove fails?

Reply:
Ok, can we fix in another PR? This is an area where we previously exited but do not now. I have no clue why this is the way it is, so let's see if we can figure that out.

Reply:
I've removed it for the time being. It does cause the `integration` tests to fail at the end, but it's pretty obvious it fails because of a umount. Still, should investigate further